### ISSUE TITLE
4.5 Issuer

### ISSUE THREAD

**Issue by nadalin** [w3c/vc-data-model#485]
The Issuer should also allowed to be a JWK

**Comment by brentzundel** [comment]
Please forgive my ignorance in this matter, is is possible for a JWK to be part of a "document containing machine-readable information about the issuer that can be used to verify the information expressed in the credential." ?

**Comment by nadalin** [comment]
Yes, I would look at https://docs.google.com/document/d/1oUDVxWy674ZWG2ElB3YN9bcmSXXR9f2mg60E_Uazigc/edit?ts=5c79597d#heading=h.13xntn55ui60 as a proposal

**Comment by brentzundel** [comment]
Thank you for that clarification, and for the document you linked to. That is from one of the groups I wish I had been able to participate in while at RWoT8.

I am not able to see anything in section 4.5 that would not allow a JWK to be used. If I have missed something, I would appreciate more clarification.

**Comment by msporny** [comment]
There is nothing in the specification that precludes `issuer` from being a JWK. For example, the following two things are legal per the data model spec:

```
  "issuer": "https://example.com/keys/foo.jwk"
```

or

```
  "issuer": {
    "id": "https://example.com/issuers/1234"
    "publicKeyJwk": {
      "kty": "RSA",
      "kid": "cc34c0a0-bd5a-4a3c-a50d-a2a7db7643df",
      "use": "sig",
      "n": "pjdss8ZaDfEH6K6U7GeW2nxDqR4IP049fk1fK0lndimbMMVBdPv_hSpm8T8EtBDxrUdi1OHZfMhUixGaut-3nQ4GG9nM249oxhCtxqqNvEXrmQRGqczyLxuh-fKn9Fg--hS9UpazHpfVAFnB5aCfXoNhPuI8oByyFKMKaOVgHNqP5NBEqabiLftZD3W_lsFCPGuzr4Vp0YS7zS2hDYScC2oOMu4rGU1LcMZf39p3153Cq7bS2Xh6Y-vw5pwzFYZdjQxDn8x8BG3fJ6j8TGLXQsbKH1218_HcUJRvMwdpbUQG5nvA2GXVqLqdwp054Lzk9_B_f1lVrmOKuHjTNHq48w",
      "e": "AQAB",
      "d": "ksDmucdMJXkFGZxiomNHnroOZxe8AmDLDGO1vhs-POa5PZM7mtUPonxwjVmthmpbZzla-kg55OFfO7YcXhg-Hm2OWTKwm73_rLh3JavaHjvBqsVKuorX3V3RYkSro6HyYIzFJ1Ek7sLxbjDRcDOj4ievSX0oN9l-JZhaDYlPlci5uJsoqro_YrE0PRRWVhtGynd-_aWgQv1YzkfZuMD-hJtDi1Im2humOWxA4eZrFs9eG-whXcOvaSwO4sSGbS99ecQZHM2TcdXeAs1PvjVgQ_dKnZlGN3lTWoWfQP55Z7Tgt8Nf1q4ZAKd-NlMe-7iqCFfsnFwXjSiaOa2CRGZn-Q",
      "p": "4A5nU4ahEww7B65yuzmGeCUUi8ikWzv1C81pSyUKvKzu8CX41hp9J6oRaLGesKImYiuVQK47FhZ--wwfpRwHvSxtNU9qXb8ewo-BvadyO1eVrIk4tNV543QlSe7pQAoJGkxCia5rfznAE3InKF4JvIlchyqs0RQ8wx7lULqwnn0",
      "q": "ven83GM6SfrmO-TBHbjTk6JhP_3CMsIvmSdo4KrbQNvp4vHO3w1_0zJ3URkmkYGhz2tgPlfd7v1l2I6QkIh4Bumdj6FyFZEBpxjE4MpfdNVcNINvVj87cLyTRmIcaGxmfylY7QErP8GFA-k4UoH_eQmGKGK44TRzYj5hZYGWIC8",
      "dp"  : "lmmU_AG5SGxBhJqb8wxfNXDPJjf__i92BgJT2Vp4pskBbr5PGoyV0HbfUQVMnw977RONEurkR6O6gxZUeCclGt4kQlGZ-m0_XSWx13v9t9DIbheAtgVJ2mQyVDvK4m7aRYlEceFh0PsX8vYDS5o1txgPwb3oXkPTtrmbAGMUBpE",
      "dq": "mxRTU3QDyR2EnCv0Nl0TCF90oliJGAHR9HJmBe__EjuCBbwHfcT8OG3hWOv8vpzokQPRl5cQt3NckzX3fs6xlJN4Ai2Hh2zduKFVQ2p-AF2p6Yfahscjtq-GY9cB85NxLy2IXCC0PF--Sq9LOrTE9QV988SJy_yUrAjcZ5MmECk",
      "qi": "ldHXIrEmMZVaNwGzDF9WG8sHj2mOZmQpw9yrjLK9hAsmsNr5LTyqWAqJIYZSwPTYWhY4nu2O0EY9G9uYiqewXfCKw_UngrJt8Xwfq1Zruz0YY869zPN4GiE9-9rzdZB33RBw8kIOquY3MK74FMwCihYx_LiU2YTHkaoJ3ncvtvg"
    }
  }
```

@nadalin, does this address the issue you initially raised?

**Comment by chaals** [comment]
It would be helpful to put an example of using an object with an `id` in. But I presumed that something like the long example above would do the job fine, based on what the spec already says.

**Comment by nadalin** [comment]
@chaals I agree as I was not thinking anyone would use the above

**Comment by brentzundel** [comment]
@nadalin, there is a PR which adds a note to the issuer section specifying that an issuer may also be a JWK. Please let us know if this addresses this issue.

**Comment by msporny** [comment]
PR #513 has been reviewed by multiple parties and merged. The next step is to see if the WG agrees with the change.

**Comment by brentzundel** [comment]
@msporny I think you may have meant PR #519

**Comment by msporny** [comment]
> @msporny I think you may have meant PR #519

My bad, yes, that's what I meant. I have fixed the comment above to refer to the appropriate PR.

**Comment by stonematt** [comment]
VCWG Teleconference Resolution: https://www.w3.org/2019/04/09-vcwg-minutes.html#resolution02

RESOLUTION: The specification is not clear with respect to the range of possible values for the issuer property and should add a few non-normative examples of different values that can be associated with the issuer property such as identifiers associated with JWKs and DIDs. Issue #485 should be closed as the PR has already been reviewed and merged.

**Comment by stonematt** [comment]
Resolved on VCWG call 4/16/2019.
RESOLVED: Issue #485 is addressed by PR #519, which makes non-normative changes noting that a JWK can also be used in the place of an `issuer` property as requested by the reviewer. The PR has been approved by multiple parties and has been merged. Issue #485 should be closed.
